node pools: add CRUD API (#17384)
hashicorp/nomad: website/content/api-docs/node-pools.mdx

---
layout: api
page_title: Node Pools - HTTP API
description: The /node/pool endpoints are used to query for and interact with node pools.
---

# Node Pools HTTP API

The `/node/pool` endpoints are used to query for and interact with node pools.

## List Node Pools

This endpoint list all node pools.

| Method | Path             | Produces           |
| ------ | ---------------- | ------------------ |
| `GET`  | `/v1/node/pools` | `application/json` |

The table below shows this endpoint's support for
[blocking queries](/nomad/api-docs#blocking-queries) and
[required ACLs](/nomad/api-docs#acls).

| Blocking Queries | ACL Required      |
| ---------------- | ----------------- |
| `YES`            | `node_pool:read`  |

### Parameters

- `prefix` `(string: "")`- Specifies a string to filter node pools based on a
  name prefix. This is specified as a query string parameter.

- `next_token` `(string: "")` - This endpoint supports paging. The `next_token`

>   parameter accepts a string which identifies the next expected node. This

Reviewer: ```suggestion
  parameter accepts a string which identifies the next expected node pool. This
```

  value can be obtained from the `X-Nomad-NextToken` header from the previous
  response.

> - `per_page` `(int: 0)` - Specifies a maximum number of nodes to return for

Reviewer: ```suggestion
- `per_page` `(int: 0)` - Specifies a maximum number of node pools to return for
```

```json
{
  "CreateIndex": 11,
  "Description": "Production workloads",
  "Meta": {
    "env": "production",
    "team": "engineering"
  },
  "ModifyIndex": 11,
  "Name": "prod-eng",
  "SchedulerConfiguration": {
    "SchedulerAlgorithm": "spread"
  }
}
```

## Create or Update Node Pool

This endpoint is used to create or update a node pool.

| Method | Path                                              | Produces           |
| ------ | ------------------------------------------------- | ------------------ |
| `POST` | `/v1/node/pool/:node_pool` <br /> `/v1/node/pool` | `application/json` |

The table below shows this endpoint's support for
[blocking queries](/nomad/api-docs#blocking-queries) and
[required ACLs](/nomad/api-docs#acls).

| Blocking Queries | ACL Required      |
| ---------------- | ----------------- |
| `NO`             | `node_pool:write` |

### Parameters

- `Name` `(string: <required>)` - Specifies the node pool to create or update.
  Must have fewer than 128 characters. Only alphanumeric, `-`, and `_` are
  allowed.

- `Description` `(string: "")` - Specifies the optional human-readable
  description of the node pool. Must have fewer than 256 characters.

>   - `Meta` `(object: null)` - Optional object with string keys and values of
>     metadata to attach to the node pool. Node pool metadata is not used by
>     Nomad and is intended for use by operators and third party tools.

Reviewer: extraneous indentation here:

```suggestion
- `Meta` `(object: null)` - Optional object with string keys and values of
  metadata to attach to the node pool. Node pool metadata is not used by
  Nomad and is intended for use by operators and third party tools.
```
